components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -520,15 +520,56 @@ public boolean validateSubscriptionDetails(String context, String version, Strin
             if (!isAdvancedThrottleEnabled) {
                 if (!defaultVersionInvoked) {
                     ps.setString(3, version);
+                    ps.setString(4, context);
+                    ps.setString(5, consumerKey);
+                    ps.setString(6, version);
                 }
             } else {
                 ps.setInt(3, apiOwnerTenantId);
                 if (!defaultVersionInvoked) {
                     ps.setString(4, version);
+                    ps.setString(5, context);
+                    ps.setString(6, consumerKey);
+                    ps.setInt(7, apiOwnerTenantId);
+                    ps.setString(8, version);
+
                 }
             }
             rs = ps.executeQuery();
-            if (rs.next()) {
+            boolean isAPISubscriptionExist = false;
+            boolean subscriptionsExist = false;
+            int productSubscriptionCount = 0;
+            while (rs.next()) {
+                subscriptionsExist = true;
+                if (isAPISubscriptionExist) {
+                    break;
+                }
+
+                // check whether the current subscription is an api subscription and if yes allow to override previous
+                // infoDTO properties
+                String productName = rs.getString("API_PRODUCT_NAME");
+
+                if (productName == null || StringUtils.isEmpty(productName)) {
```

Comment:
don't hv to do a null check because StringUtils.isEmpty does this

Reply:
fixed in https://github.com/wso2/carbon-apimgt/pull/6210/commits/2a3722362e699478b9632e859503e3939f3036af